s3_tagger.py

```diff
@@ -80,60 +80,78 @@ def read_csv(csv_location, s3_client):
+    attempt_to_tag = False
+
+    if split_string[-1].endswith("_$folder$"):
+        split_string[-1] = split_string[-1][0:-9]
+
+    if split_string[-2] in csv_data:
+        db_name = split_string[-2]
+        table_name = split_string[-1]
+        attempt_to_tag = True
+
+    elif split_string[-3] in csv_data:
         db_name = split_string[-3]
+        table_name = split_string[-2]
+        attempt_to_tag = True
+
+    elif split_string[-4] in csv_data:
+        db_name = split_string[-4]
+        table_name = split_string[-3]
+        attempt_to_tag = True
+
     else:
-        table_name = split_string[-1][0:-9]
-        db_name = split_string[-2]
+        logger.warning(
+            f'Couldn\'t establish a valid database and table name for key ", "table_name": "", "db_name": "", "key": "{key}"'
+        )
 
     if db_name.endswith(".db"):
         db_name = db_name.replace(".db", "")
+
     pii_value = ""
     tag_info_found = False
-    db_missing_from_csv = False
 
-    if db_name in csv_data:
+    if attempt_to_tag:
         for table in csv_data[db_name]:
             if table_name == table["table"]:
                 pii_value = table["pii"]
                 tag_info_found = True
-    else:
-        db_missing_from_csv = True
 
     if type(pii_value) != str:
         pii_value = ""
 
-    if db_missing_from_csv:
-        logger.warning(
-            f'Database is missing from the CSV data ", "table_name": "{table_name}", "db_name": "{db_name}'
-        )
-
     if not tag_info_found:
         logger.warning(
-            f'Table is missing from the CSV data ", "table_name": "{table_name}", "db_name": "{db_name}'
+            f'Table is missing from the CSV data ", "table_name": "{table_name}", "db_name": "{db_name}", "key": "{key}"'
         )
 
     elif pii_value == "":
         logger.info(
-            f'No PII value as the table has yet to be classified ", "table_name": "{table_name}", "db_name": "{db_name}'
+            f'No PII value as the table has yet to be classified ", "table_name": "{table_name}", "db_name": "{db_name}", "key": "{key}"'
         )
 
-    try:
-        s3_client.put_object_tagging(
-            Bucket=s3_bucket,
-            Key=key,
-            Tagging={
-                "TagSet": [
-                    {"Key": "db", "Value": db_name},
-                    {"Key": "table", "Value": table_name},
-                    {"Key": "pii", "Value": pii_value},
-                ]
-            },
+    if attempt_to_tag:
+        try:
+            s3_client.put_object_tagging(
+                Bucket=s3_bucket,
+                Key=key,
+                Tagging={
+                    "TagSet": [
+                        {"Key": "db", "Value": db_name},
+                        {"Key": "table", "Value": table_name},
+                        {"Key": "pii", "Value": pii_value},
+                    ]
+                },
+            )
+            logger.info(f'Successfully tagged", "object": "{key}')
```

Comment:
Not keen on logging out we have tagged a file. We are running this against 600K files per day, thats alot of logging! Can we only log when we fail?

Reply:
It's great for debugging - I would prefer to keep it. It does no harm

Reply:
I would suggest you haven't tried reading a 600K log file before and thought it was great for debugging? ;-) You do realise that searching for Successfully tagged is the same as searching for Failed to tag, and not finding anything.